### ISSUE TITLE
Add namespace as optional param for items in namespaceResourceBlacklist via AppProject

### ISSUE THREAD

**Issue by zbialik** [argoproj/argo-cd#9047]
# Summary

We enable our developers to use ArgoCD for gitops deploys to k8s clusters that we provision for them. We provide our developers an AppProject to use to deploy their k8s resources to their provided k8s cluster(s).

**We want the ability to restrict them from deploying any resources to the 'kube-system' namespace.**

The AppProject only allows us to restrict the **types of resources** they can deploy, but not **what namespaces** they can deploy.

# Motivation

Kubernetes Admins should be able to manage the core-infrastructure of k8s clusters using the App of Apps pattern, while enabling developers to do the same for their app-specific resources. 

As of right now, we are forced to choose between allowing them the ability to create (or break) resources in 'kube-system' namespace OR granularly setting the namespaces in the `spec.destinations` list of the AppProject manifests. 

- We do NOT want to be gatekeepers for these clusters so having to granularly allow namespaces for our developers to deploy to is not ideal. We'd prefer to blacklist these namespaces and allow them to create/deploy to anywhere else.

# Proposal

I would like to have the ability to set a `namespace` field as part of an item in `spec.namespaceResourceBlacklist` list.

Something like:

```
apiVersion: argoproj.io/v1alpha1
kind: AppProject
metadata:
  name: team-a
spec:
  description: "Example project"
  sourceRepos:
  - 'https://example.local/team-a/kubernetes.git'
  destinations:
  - namespace: '*'
    server: 'https://team-a.example.kubernetes'
  namespaceResourceBlacklist:
  - group: ''
    kind: ''
    namespace: 'kube-system'
```

Alternatively, the `spec.destinations` could be enhanced to include some sorta namespaceBlacklist of its own. Not sure which is easier to implement.

**Comment by rishabh625** [comment]
@ishitasequeira : this change will belong to component-api ? also looks an easy enhancement to add restrict namespace deployment

**Comment by crenshaw-dev** [comment]
Someone suggested doing this as a type of pattern in the `destinations` field. https://github.com/argoproj/argo-cd/issues/9464

**Comment by blakepettersson** [comment]
Indeed, this should be addressed with #9652 😄
